### ISSUE TITLE
bug: IonReorder event outputs different "to" value based on how quickly you drag.

### ISSUE THREAD

**Issue by ebk46** [ionic-team/ionic-framework#19860]
# Bug Report

**Ionic version:**
[x] **4.11.3 (React)**

**Current behavior:**
Reordering an item down with IonReorder provides different "to" values based on how fast the item is dragged. If you drag too quickly, the value is 1 higher than it should be, presumably because it hasn't had time to fully "remove" the item from its original position.

**Expected behavior:**
The "to" value should always be the same, no matter how quickly the item is dragged to that position.

**Steps to reproduce:**
See StackBlitz for this example. 

Assuming a 3 item list:

1. Drag item from top position to bottom position slowly. Note that the change should be from 0 to 2, as 0 is the first element, and 2 is the last.

2. Drag item from top position to bottom position QUICKLY. Note that the change SHOULD be from 0 to 2 again, but is instead 0 to 3 because WHILE the item is being dragged, it still considers that item in the list in its original order.

**Related code:**
https://stackblitz.com/edit/ionic-v4-11-3-react-reorder?file=Page1.jsx

**Other information:**

**Ionic info:**
```
Ionic:

   Ionic CLI       : 5.4.5 (/Users/evan/.config/yarn/global/node_modules/ionic)
   Ionic Framework : @ionic/react 4.11.3

Capacitor:

   Capacitor CLI   : 1.3.0
   @capacitor/core : 1.3.0

Utility:

   cordova-res : 0.6.0 (update available: 0.8.1)
   native-run  : not installed

System:

   NodeJS : v12.10.0 (/usr/local/Cellar/node/12.10.0/bin/node)
   npm    : 6.11.3
   OS     : macOS High Sierra
```

**Comment by ebk46** [comment]
I tested this some more and it appears to only happen when moving things to the bottom of the list. Not 100% sure though.

**Comment by liamdebeasi** [comment]
Thanks for the issue. This bug was resolved via https://github.com/ionic-team/ionic-framework/pull/23797 and a fix is available now in Ionic React 5.7.0 or newer.
